### PR TITLE
Smart Contract fucntionality ready with required functions, Contains …

### PR DESCRIPTION
**Pull request by Jasj0t**
…3 extra fucntions for V2 of this project

**Description**

1. Added hardhat, solidity and openzeppelin
2. This modified the readme file and added a few lines to it
3. extra folders made by hardhat when you initialize a smart contract, a test.js file, a contracts folder where the smart contract code is available, plus a default test for the smart contract.

**Trello**

- Link: https://trello.com/c/GhDXFgW0/54-f3-smart-contract-generation-7

**Extra Info**

Their are a total of 7 functions within the contact/marketplace.js file.
4 of these are as per requirements of sprint 1
3 of these are made a part of v2 of this project which include functions related to sale purchase and a commission for making smart contracts. For now I have made the listing price as 0 ether which will be changed in accordance to decisions taken by  entire team for v2.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,27 @@ node_modules/
 dist/
 .env
 .nyc_output/
-/coverage
\ No newline at end of file
+/coverage
```
Comment (xuqun123) [inline .gitignore:7]: we don't need any files from `/coverage` folder to be uploaded to this repo.

Please remove this change.

File: .gitignore [hunk 2]
```diff
@@ -4,4 +4,27 @@ node_modules/
...
+node_modules
+.env
+coverage
+coverage.json
+typechain
+typechain-types
+
+# Hardhat files
+cache
+artifacts
```
Comment (xuqun123) [inline .gitignore]: Please remove these duplicate entries. They're added from above already.

File: .gitignore [hunk 3]
```diff
@@ -4,4 +4,27 @@ node_modules/
 dist/
 .env
 .nyc_output/
-/coverage
\ No newline at end of file
+/coverage
+node_modules
+.env
+coverage
```
Comment (xuqun123) [inline .gitignore]: Remove this line, it's included above already.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,17 @@
 # crazy-bee-project
 
 ![Coverage](./frontend/coverage-badge.svg) ![Coverage](./backend/coverage-badge.svg)
+
+# Sample Hardhat Project
```
Comment (xuqun123) [inline README.md]: We don't want to document a sample project in the main README doc. 
Please move these new changes info into a new `.md` file inside the `/docs` folder.

This main `README` needs to summarise all finalised functionalities/services/applications we have for this project or introduce the project in general.

File: README_1.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# crazy-bee-project
+
+![Coverage](./frontend/coverage-badge.svg) ![Coverage](./backend/coverage-badge.svg)
```
Comment (xuqun123) [inline README_1.md]: This file is not needed. Delete this one, pls!

File: contracts/marketplace.sol [hunk 6]
```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: MIT
```
Comment (xuqun123) [inline contracts/marketplace.sol]: Please move all your NFT-related changes into a new folder like `nft` or `nft-minting` (including all of the `package.json` and `package-lock.json` file changes from this PR -> you can create a new `package.json` file from your new folder)

We don't want to break the current folder structure for the project.

The project's folder structure should be like:
```
/crazy-bee-project
  - /docs
  - /frontend
  - /backend
  - /ai-creator
  - /nft-minting

```

**Comment by xuqun123** [comment]
Please also fix your PR title to match the same pattern as other PRs. 

e.g., https://github.com/xuqun123/crazy-bee-project/pull/9

**Comment by Jasj0t** [comment]
I have made the appropriate changes and committed them please review.

**Review by xuqun123 — APPROVED** [review]

**Pull request merged**